### PR TITLE
Move js to vue

### PR DESCRIPTION
**Pull request by OmgImAlexis**
This moves the js to the vue mounted for schedule, history and manage_massEdit.

### CODE REVIEW

File: themes-default/slim/views/schedule.mako [hunk 1]
```diff
@@ -19,6 +19,82 @@ const startVue = () => {
         el: '#vue-wrap',
         data() {
             return {};
+        },
+        mounted() {
+            if ($.isMeta({ layout: 'schedule' }, ['list'])) {
+                const sortCodes = {
+                    date: 0,
```
Comment (p0psicles) [inline themes-default/slim/views/schedule.mako]: Switching sorting breaks the page.

image

File: themes-default/slim/views/history.mako [hunk 2]
```diff
@@ -21,6 +21,56 @@ const startVue = () => {
         el: '#vue-wrap',
         data() {
             return {};
+        },
+        mounted() {
+            $('#historyTable:has(tbody tr)').tablesorter({
```
Comment (p0psicles) [inline themes-default/slim/views/history.mako]: Tries to load index.js.
GET http://localhost:8081/js/history/index.js?10032 404 (Not Found)

File: themes-default/slim/views/schedule.mako [hunk 3]
```diff
@@ -19,6 +19,82 @@ const startVue = () => {
...
+                    show: 2,
```
Comment (p0psicles) [inline themes-default/slim/views/schedule.mako]: GET http://localhost:8081/js/schedule/index.js?10032 net::ERR_ABORTED
Error on load of index.js.

File: themes-default/slim/views/schedule.mako (hunk 1 above)
Comment (OmgImAlexis) [reply themes-default/slim/views/schedule.mako]: I'll fix this.

File: themes-default/slim/views/manage_massEdit.mako [hunk 4]
```diff
@@ -15,14 +15,49 @@
     allowed_qualities, preferred_qualities = common.Quality.split_quality(initial_quality)
 %>
 <script type="text/javascript" src="js/quality-chooser.js?${sbPID}"></script>
-<script type="text/javascript" src="js/mass-edit.js?${sbPID}"></script>
 <script>
 let app;
 const startVue = () => {
     app = new Vue({
```
Comment (p0psicles) [inline themes-default/slim/views/manage_massEdit.mako]: Warning while trying to load Vue.
```
vender.min.js?10032:2 jQuery.Deferred exception: Cannot set property '_renderItem' of undefined TypeError: Cannot set property '_renderItem' of undefined
    at ma.fn.init.$.fn.fileBrowser (http://localhost:8081/js/browser.js?10032:169:52)
    at Vue$3.mounted (http://localhost:8081/manage/massEdit:962:28)
    at callHook (http://localhost:8081/js/lib/vue.js:1802:19)
    at Vue$3.Vue._mount (http://localhost:8081/js/lib/vue.js:1676:7)
    at Vue$3.$mount (http://localhost:8081/js/lib/vue.js:5506:15)
    at Vue$3.$mount (http://localhost:8081/js/lib/vue.js:7381:16)
    at initRender (http://localhost:8081/js/lib/vue.js:2173:8)
    at Vue$3.Vue._init (http://localhost:8081/js/lib/vue.js:2515:5)
    at new Vue$3 (http://localhost:8081/js/lib/vue.js:2557:8)
    at startVue (http://localhost:8081/manage/massEdit:925:11) undefined
```
image

Comment (p0psicles) [reply themes-default/slim/views/manage_massEdit.mako]: This is the massEdit page.

File: themes-default/slim/views/schedule.mako (hunk 1 above)
Comment (p0psicles) [reply themes-default/slim/views/schedule.mako]: This should be fixed now

File: themes-default/slim/views/manage_massEdit.mako [hunk 5]
```diff
@@ -15,14 +15,49 @@
...
+
+            $('.delete_root_dir').on('click', function() {
+                const curIndex = findDirIndex($(this).attr('id'));
+                $('#new_root_dir_' + curIndex).val(null);
+                $('#display_new_root_dir_' + curIndex).html('<b>DELETED</b>');
+            });
+
+            $('#location').fileBrowser({
+                title: 'Select Show Location'
+            });
```
Comment (sharkykh) [inline themes-default/slim/views/manage_massEdit.mako]: From what I can see, this is what is breaking the page. **And this is not needed.**
I assume this is supposed to set the title for the folder browser when you click on Edit.
Remove this and instead add:
`title: 'Select Show Location'` to the object on lines 46-49:
```js
                $(this).nFileBrowser(editRootDir, {
                    initialDir,
					title: 'Select Show Location',
                    whichId: curIndex
                });
```
That fixed the rendering of the page for me. Folder browser tested and working.

File: themes-default/slim/views/schedule.mako (hunk 3 above)
Comment (sharkykh) [reply themes-default/slim/views/schedule.mako]: From what I see, the import just needs to be removed from `main.mako`.
https://github.com/pymedusa/Medusa/blob/bca8f0c03547625646faec742988e192855e9d97/themes-default/slim/views/layouts/main.mako#L96

File: themes-default/slim/views/history.mako (hunk 2 above)
Comment (sharkykh) [reply themes-default/slim/views/history.mako]: From what I see, the import just needs to be removed from `main.mako`.
https://github.com/pymedusa/Medusa/blob/bca8f0c03547625646faec742988e192855e9d97/themes-default/slim/views/layouts/main.mako#L116

File: themes-default/slim/views/manage_massEdit.mako (hunk 5 above)
Comment (p0psicles) [reply themes-default/slim/views/manage_massEdit.mako]: Tnx @sharkykh. I've committed it.

**Comment by p0psicles** [comment]
I think the only issue left, is the call for the old js.

File: themes-default/slim/views/history.mako (hunk 2 above)
Comment (OmgImAlexis) [reply themes-default/slim/views/history.mako]: @sharkykh @p0psicles the js was moved into that mounted method and I must have missed removing the script tag.

File: themes-default/slim/views/schedule.mako (hunk 3 above)
Comment (OmgImAlexis) [reply themes-default/slim/views/schedule.mako]: Same as other script tag, can be removed.

File: themes-default/slim/views/manage_massEdit.mako (hunk 4 above)
Comment (OmgImAlexis) [reply themes-default/slim/views/manage_massEdit.mako]: @p0psicles was this sorted?

**Comment by OmgImAlexis** [comment]
@p0psicles @medariox @sharkykh can we merge?

File: themes-default/slim/views/manage_massEdit.mako (hunk 4 above)
Comment (medariox) [reply themes-default/slim/views/manage_massEdit.mako]: None of the buttons in Mass Update seem to work for me (`/manage`).

**Comment by p0psicles** [comment]
I think it should be fully tested some more. As we don't want to add more defects to develop as that we already have.

**Comment by sharkykh** [comment]
Does the link to edit selected shows on Mass Update page work for you?

**Comment by OmgImAlexis** [comment]
@medariox @sharkykh please confirm the buttons work.

**Comment by sharkykh** [comment]
https://github.com/sharkykh/Medusa/commit/a4613e76d6c1e1fd6bf1f4a1ccb494412bb59eb1 ([patch](https://github.com/sharkykh/Medusa/commit/a4613e76d6c1e1fd6bf1f4a1ccb494412bb59eb1.patch), [diff](https://github.com/sharkykh/Medusa/commit/a4613e76d6c1e1fd6bf1f4a1ccb494412bb59eb1.diff))
^ Closes #3943. The commit is based on this branch.

**Comment by OmgImAlexis** [comment]
@p0psicles if you want to approve and merge this we can get #3947 tested and merged.

**Comment by OmgImAlexis** [comment]
We almost need to disable codebeat on the themes and themes-default dirs.

**Pull request merged**